app/makers_bnb.rb

```diff
@@ -0,0 +1,112 @@
+ENV["RACK_ENV"] ||= "development"
+
+require 'sinatra/base'
+require 'sinatra/flash'
+require_relative 'data_mapper_setup'
+
+class MakersBnB < Sinatra::Base
+  enable :sessions
+  register Sinatra::Flash
+  use Rack::MethodOverride
+  set :session_secret, 'super secret'
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id])
+    end
+  end
+
+  post '/spaces/filter' do
+    session[:filter_from] = params[:filter_from]
+    session[:filter_to] = params[:filter_to]
```

Comment:
I would pass the `filter` method down into the model. **Always look for ways to push things further down the stack.** In addition, I would do this in a GET request, because we're not _changing_ anything on the model. We're just GETting a subset of them. So this would read something like

```ruby
get '/spaces/filter' do
  @spaces = Space.filter(params)
  erb :homepage
end
```